tmail-backend/mailets/src/main/java/com/linagora/tmail/mailet/LDAPMailingList.java

```diff
@@ -449,18 +457,41 @@ SenderValidationPolicy domainRestrictedValidationPolicy() {
        };
    }
 
-   SenderValidationPolicy chooseSenderValidationPolicy(SearchResultEntry list) {
-       Attribute businessCategory = list.getAttribute("businessCategory");
-       if (businessCategory == null) {
-           return SenderValidationPolicy.OPEN;
-       }
-       return switch (businessCategory.getValue().toLowerCase().trim()) {
-           case "openlist" -> SenderValidationPolicy.OPEN;
-           case "internallist" -> SenderValidationPolicy.INTERNAL.apply(getMailetContext());
-           case "memberrestrictedlist" -> memberSenderValidationPolicy();
-           case "ownerrestrictedlist" -> ownerSenderValidationPolicy();
-           case "domainrestrictedlist" -> domainRestrictedValidationPolicy();
-           default -> SenderValidationPolicy.OPEN;
-       };
-   }
+    SenderValidationPolicy chooseSenderValidationPolicy(SearchResultEntry list) {
+        Attribute businessCategory = list.getAttribute("businessCategory");
+        if (businessCategory == null) {
+            return SenderValidationPolicy.OPEN;
+        }
+        return switch (extractBusinessCategory(businessCategory.getValue().toLowerCase().trim())) {
+            case "openlist" -> SenderValidationPolicy.OPEN;
+            case "internallist" -> SenderValidationPolicy.INTERNAL.apply(getMailetContext());
+            case "memberrestrictedlist" -> memberSenderValidationPolicy();
+            case "ownerrestrictedlist" -> ownerSenderValidationPolicy();
+            case "domainrestrictedlist" -> domainRestrictedValidationPolicy();
+            default -> SenderValidationPolicy.OPEN;
+        };
+    }
+
+    private String extractBusinessCategory(String ldapValue) {
+        return businessCategoryExtractAttribute
+            .map(extractAttribute -> santinizeBusinessCategory(ldapValue, extractAttribute))
+            .orElse(ldapValue);
+    }
+
+    private String santinizeBusinessCategory(String ldapValue, String extractAttribute) {
+        if (ldapValue != null && !ldapValue.isEmpty()) {
+            // Split the businessCategory string by comma, map to key-value pairs, and collect to a MultiMap<String, String>
+            ImmutableListMultimap<String, String> businessAttributeMap = Splitter.on(',')
+                .trimResults()
+                .splitToStream(ldapValue)
+                .map(part -> part.trim().split("=", 2))
+                .filter(keyValue -> keyValue.length == 2) // Ensure valid key-ldapValue pairs
+                .collect(ImmutableListMultimap.toImmutableListMultimap(keyValue -> keyValue[0].trim(), keyValue -> keyValue[1].trim()));
```

Comment:
Use unboundid DN class (or something similar) for the parsing?

Reply:
```java
    private String santinizeBusinessCategory(String ldapValue) throws LDAPException {
        if (ldapValue != null && !ldapValue.isEmpty()) {
            DN dn = new DN(ldapValue);

            // Iterate through the RDNs (Relative Distinguished Names)
            for (RDN rdn : dn.getRDNs()) {
                // Iterate over the attribute names within the RDN
                String[] attributeNames = rdn.getAttributeNames();
                String[] attributeValues = rdn.getAttributeValues();

                for (int i = 0; i < attributeNames.length; i++) {
                    if ("cn".equalsIgnoreCase(attributeNames[i])) {
                        return attributeValues[i];
                    }
                }
            }
        }
        return "";
    }
```

Works